### PR TITLE
Base components/styling

### PR DESCRIPTION
**Pull request by tawnyzhao**
Creates the base components for the storefront and team dashboard. 
Uses a global ThemeProvider to pass color and typography options.
Creates common base components (buttons, icons, etc). 

Example (with global theming): 
image

### CODE REVIEW

**Comment by tawnyzhao** [comment]
Dashboard button
image
Dashboard text input
image

Storefront stuff will need to be another PR

File: src/shared/App.js [hunk 1]
```diff
@@ -35,15 +38,21 @@ function App() {
               console.log('The team', team);
               return (
                 <React.Fragment key={i}>
```
Comment (tawnyzhao) [inline src/shared/App.js:48]: why are we mapping over all teams to make the routes here?

**Review by alexguo8 — APPROVED** [review]
Looks good to me!

**Review by faizaanmadhani — APPROVED** [review]
This looks good! Just make sure to fix the merge conflicts.

**Pull request merged**